### DISCUSSION TITLE
diarize: how to

### DISCUSSION THREAD

**Discussion by amohajerani** [deepgram/community#306] [General help] [answered]
I am following the [example flask project](https://github.com/deepgram/deepgram-python-sdk/tree/main/sample-projects/streaming-audio/Flask/live-transcription-flask) for transcribing streamed audio. I have [set](https://github.com/amohajerani/deepgram-tutorial/blob/main/main.py#L32) `'diarize':True`, but the output does not have any of the fields that a diarized transcription should have (no speaker field for instance).
[This](https://github.com/amohajerani/deepgram-tutorial) is my code.

Here is an example of the logged message on the browser's console.
How can I get the diarized transcription?

```
MessageEvent {isTrusted: true, data: 'In a world.', origin: 'ws://localhost:5555', lastEventId: '', source: null, …}
isTrusted
: 
true
bubbles
: 
false
cancelBubble
: 
false
cancelable
: 
false
composed
: 
false
currentTarget
: 
WebSocket {url: 'ws://localhost:5555/listen', readyState: 1, bufferedAmount: 0, onerror: null, onopen: ƒ, …}
data
: 
"In a world."
defaultPrevented
: 
false
eventPhase
: 
0
lastEventId
: 
""
origin
: 
"ws://localhost:5555"
ports
: 
[]
returnValue
: 
true
source
: 
null
srcElement
: 
WebSocket {url: 'ws://localhost:5555/listen', readyState: 1, bufferedAmount: 0, onerror: null, onopen: ƒ, …}
target
: 
WebSocket {url: 'ws://localhost:5555/listen', readyState: 1, bufferedAmount: 0, onerror: null, onopen: ƒ, …}
timeStamp
: 
13950.100000023842
type
: 
"message"
userActivation
: 
null
[[Prototype]]
: 
MessageEvent
```

**Comment by jjmaldonis** [comment]
Hey @amohajerani this is because the `index.html` writes the `data` field of the response to the webpage. See this line of code in the example flask project: https://github.com/deepgram/deepgram-python-sdk/blob/main/sample-projects/streaming-audio/Flask/live-transcription-flask/templates/index.html#L35  You can modify the HTML/JS to output the data you need.

**Reply by amohajerani** [reply]
Thanks @jjmaldonis. Appreciate your help. I did a console.log(message) and confirmed the `speaker` field does not exist in the message. So the HTML is not the issue.  I pasted the console.log output on my original post (above)
Here is the my code: https://github.com/amohajerani/deepgram-tutorial/blob/main/templates/index.html#L33

**Reply by jjmaldonis** [reply, marked as answer]
Got it, thanks for the link to your code. I missed that this example does not send the full message from the python backend to the HTML/JS frontend.

On this line: https://github.com/amohajerani/deepgram-tutorial/blob/main/main.py#L24
change `await fast_socket.send_str(transcript)` to `await fast_socket.send_str(data)`. That will send the entire message, which you can then parse in the JS. You may also need to change the JS in the index.html file.
